### PR TITLE
Add infrastructure files

### PR DESCRIPTION
**Pull request by mrakitin**
Summary:
- add a license file
- add versioneer to correctly deal with versions based on tags
- add `MANIFEST.in` for source distribution
- extract dependency into `requirements.txt`
- update `setup.py` based on https://github.com/NSLS-II/nslsii/blob/master/setup.py (to be updated to use https://github.com/NSLS-II/scientific-python-cookiecutter/blob/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D/setup.py)

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,29 @@
+BSD 3-Clause License
+
+Copyright (c) 2020, Brookhaven Science Associates, Brookhaven National Laboratory
```
Comment (MikeHart85) [inline LICENSE]: This does not need to name BSA and should just name BNL, like so:

`Copyright (c) 2020, Brookhaven National Laboratory`

See @stuartcampbell's comment [on this issue](https://github.com/NSLS-II/scientific-python-cookiecutter/issues/89#issuecomment-591649337).

Comment (mrakitin) [reply LICENSE]: OK, thanks for pointing it out. I actually copied it from SRX, I'll correct it there as well.

**Comment by mrakitin** [comment]
This is done for now. I tested the code correctly handles the versions of the package based on the tag/number of commits from tag.

CI can be enabled in a separate PR.

**Comment by mrakitin** [comment]
Please note this commit: 5466d2e. It's only needed when we render `README.md` files. We don't need it for plain text or `README.rst` files (which are default in https://github.com/NSLS-II/scientific-python-cookiecutter).

xref: https://github.com/NSLS-II/scientific-python-cookiecutter/issues/25.

File: setup.py [hunk 2]
```diff
@@ -1,39 +1,62 @@
...
     author="Brookhaven National Laboratory",
+    author_email="DAMA@bnl.gov",
+    url="https://github.com/NSLS-II-HXN/hxntools",
+    python_requires='>={}'.format('.'.join(str(n) for n in min_version)),
+    packages=find_packages(exclude=['docs', 'tests']),
+    include_package_data=True,
     install_requires=requirements,
+    license="BSD-3-Clause",
+    classifiers=[
+        "Development Status :: 2 - Pre-Alpha",
```
Comment (jklynch) [inline setup.py:60]: Should the development status be a little higher?

File: setup.py [hunk 3]
```diff
@@ -1,24 +1,64 @@
...
+    author="Brookhaven National Laboratory",
+    author_email="DAMA@bnl.gov",
+    url="https://github.com/NSLS-II-HXN/hxntools",
+    python_requires='>={}'.format('.'.join(str(n) for n in min_version)),
+    packages=find_packages(exclude=['docs', 'tests']),
+    include_package_data=True,
+    install_requires=requirements,
+    license="BSD-3-Clause",
+    classifiers=[
+        "Development Status :: 2 - Pre-Alpha",
```
Comment (jklynch) [inline setup.py]: Should the development status be a little higher? `Development Status :: 5 - Production/Stable` seems reasonable.

**Review by jklynch — CHANGES_REQUESTED** [review]

**Review by MikeHart85 — APPROVED** [review]
Agree with Josh's comment. Everything else looks good to go.

File: setup.py (hunk 3 above)
Comment (mrakitin) [reply setup.py]: Updated via https://github.com/NSLS-II-HXN/hxntools/pull/21/commits/1c1247057b4c2bc34d3500f7b5f4b9fc2debb361.

**Review by jklynch — APPROVED** [review]

**Comment by mrakitin** [comment]
Thank you!

**Pull request merged**